### PR TITLE
Externalize JavaScript: export list

### PR DESCRIPTION
**Pull request by orangejenny**
Review by commit, skipping the first two (because they're from https://github.com/dimagi/commcare-hq/pull/15691).

@esoergel / @millerdev

### CODE REVIEW

File: corehq/apps/dashboard/static/dashboard/js/dashboard_domain.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+hqDefine('dashboard/js/dashboard_domain.js', function() {
+    'use strict';
+    var initial_page_data = hqImport('hqwebapp/js/initial_page_data.js').get;
+    var dashboardApp = window.angular.module('dashboardApp', [
+        'ui.bootstrap',
+        'ngResource',
+        'ngRoute',
+        'ng.django.rmi',
+        'hq.dashboard'
```
Comment (dimagimon) [inline corehq/apps/dashboard/static/dashboard/js/dashboard_domain.js]: Missing trailing comma. (comma-dangle)

File: corehq/apps/export/static/export/js/export_list.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+hqDefine("export/js/export_list.js", function() {
+   'use strict';
```
Comment (dimagimon) [inline corehq/apps/export/static/export/js/export_list.js]: Expected indentation of 4 space characters but found 3. (indent)

File: corehq/apps/export/static/export/js/export_list.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    listExportsApp.constant('legacy_bulk_download_url', initial_page_data("legacy_bulk_download_url"));
+    listExportsApp.constant('modelType', initial_page_data("model_type"));
+    listExportsApp.constant('staticModelType', initial_page_data("static_model_type"));
+    listExportsApp.constant('filterFormElements', {
+        emwf_form_filter: function () {
+            return $('#id_emwf_form_filter');
+        },
+        emwf_case_filter: function () {
+            return $('#id_emwf_case_filter');
+        }
```
Comment (dimagimon) [inline corehq/apps/export/static/export/js/export_list.js]: Missing trailing comma. (comma-dangle)

File: corehq/apps/hqwebapp/views.py [hunk 4]
```diff
@@ -1194,3 +1195,18 @@ def couch_doc_counts(request, domain):
...
+    # Add the output of djng_current_rmi to view context, which requires having
+    # the rest of the context, specifically context['view'], available.
+    # See https://github.com/jrief/django-angular/blob/master/djng/templatetags/djng_tags.py
+    def get_context_data(self, **kwargs):
+        context = super(HQJSONResponseMixin, self).get_context_data(**kwargs)
+        from djangular.templatetags.djangular_tags import djng_current_rmi
+        context['djng_current_rmi'] = json.loads(djng_current_rmi(context))
+        return context
+
+
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/views.py]: W391 blank line at end of file

**Review by esoergel — APPROVED** [review]
Very easy to review, thanks for the commit-by-commit breakdown!

**Review by millerdev — APPROVED** [review]

**Pull request merged**